aea/registries/base.py

```diff
@@ -533,3 +546,76 @@ def teardown(self):
         """
         for r in self._registries:
             r.teardown()
+
+
+class Filter(object):
+    """This class implements the filter of an AEA."""
+
+    def __init__(self, resources: Resources, decision_maker_out_queue: Queue):
+        """
+        Instantiate the filter.
+
+        :param resources: the resources
+        :param decision_maker_out_queue: the decision maker queue
+        """
+        self._resources = resources
+        self._decision_maker_out_queue = decision_maker_out_queue
+        # TODO: self._inactive_handlers = {}  # type: Dict[SkillId, List[HandlerId]]
```

Comment:
Note, @marcofavorito how this demonstrates the need for `HandlerId`, `BehaviourId` and `TaskId`